src/linker/Linker/LinkContext.cs

```diff
@@ -663,6 +663,169 @@ public TargetRuntimeVersion GetTargetRuntimeVersion ()
 
 			return _targetRuntime.Value;
 		}
+
+		readonly Dictionary<MethodReference, MethodDefinition> methodresolveCache = new ();
+		readonly Dictionary<FieldReference, FieldDefinition> fieldresolveCache = new ();
+		readonly Dictionary<TypeReference, TypeDefinition> typeresolveCache = new ();
+
+		public MethodDefinition ResolveMethodDefinition (MethodReference methodReference)
+		{
+			if (methodReference is MethodDefinition md)
+				return md;
+
+			if (methodReference is null)
+				return null;
+
+			if (methodresolveCache.TryGetValue (methodReference, out md)) {
+				if (md == null && !IgnoreUnresolved)
+					ReportUnresolved (methodReference);
```

Comment:
I wonder if we should cache that we already reported the unresolved reference - to avoid repeating the same warning multiple times. We could store a sentinel value in the cache as a mark of that happening.

Reply:
Added extra cache for that